torch/distributed/distributed_c10d.py

```diff
@@ -519,14 +531,35 @@ class GroupMember(metaclass=_WorldMeta):
 STORE_BASED_BARRIER_PREFIX = "store_based_barrier_key"
 
 
-def _get_pg_device(group: ProcessGroup):
+def _get_object_coll_device(group: Optional[ProcessGroup] = None):
+    group = group or _get_default_group()
+    """
+    ``group._device_types`` is a property pybind that returns the devices
+    ("cpu", "cuda", etc) supported by ``group``. Can be multiple if the
+    ``group`` supports multiple devices.
+    """
+    devices = group._device_types
+
+    if len(devices) == 1:
+        # User fixed exactly one backend in `init_process_group`
+        return devices[0]
+    elif len(devices) == 0:
+        # No backend has been registered with this PG (maybe because no
+        # collective has been run?) We pick cpu as the default and hopefully
+        # this would lazily init Gloo or other available cpu backend.
```

Comment:
is it true that this would today lazily init gloo?

Reply:
Not the case today. But hopefully (say Gloo is improved or other CPU backend implements that way).
This `elif` condition is not hit today because `len(devices)` is always > 0 per today's init code. But just adding code here to be future safe.